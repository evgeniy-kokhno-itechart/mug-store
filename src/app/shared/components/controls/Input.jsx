import React from 'react';
import { PropTypes } from 'prop-types';
import FormGroup from '../markup/FormGroup';
import ControlBase from './ControlBase';

class Input extends ControlBase {
  render() {
    const {
      name, label, error, type, value,
    } = this.props;
    return (
      <FormGroup name={name} label={label} error={error}>
        <input type={type} value={value} key={name} name={name} id={name} onChange={this.handleChange} className="form-control" />
      </FormGroup>
    );
  }
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Input.defaultProps = {
  type: 'text',
  error: '',
  value: '',
};

export default Input;
