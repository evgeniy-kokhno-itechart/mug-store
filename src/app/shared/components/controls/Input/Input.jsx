import React from 'react';
import { PropTypes } from 'prop-types';
import FormGroup from '../../markup/FormGroup/FormGroup';
import ControlBase from '../ControlBase';
import './Input.scss';

class Input extends ControlBase {
  render() {
    const {
      name, label, error, type, value,
    } = this.props;
    return (
      <FormGroup name={name} label={label} error={error}>
        <input
          type={type}
          value={value}
          key={name}
          name={name}
          id={name}
          onChange={this.handleChange}
          className="formgroup__input focusable"
        />
      </FormGroup>
    );
  }
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Input.defaultProps = {
  type: 'text',
  error: '',
  value: '',
};

export default Input;
