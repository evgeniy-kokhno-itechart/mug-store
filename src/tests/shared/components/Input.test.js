/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Input from '../../../app/shared/components/controls/Input/Input';

configure({ adapter: new Adapter() });

describe('<Input />', () => {
  let InputWrapper;

  beforeEach(() => {
    InputWrapper = shallow(<Input name="testname" label="Test label" />);
  });

  it('check Input renders properly with default props', () => {
    expect(InputWrapper).toMatchSnapshot();
  });

  it('check Input renders properly with error', () => {
    InputWrapper.setProps({ error: 'test error' });
    expect(InputWrapper).toMatchSnapshot();
  });

  it('check Input renders properly with custom type provided', () => {
    InputWrapper.setProps({ type: 'password' });
    expect(InputWrapper).toMatchSnapshot();
  });

  it('check Input renders properly with string value provided', () => {
    InputWrapper.setProps({ value: 'Test value' });
    expect(InputWrapper).toMatchSnapshot();
  });

  it('check Input renders properly with number value provided', () => {
    InputWrapper.setProps({ value: 5 });
    expect(InputWrapper).toMatchSnapshot();
  });
});
