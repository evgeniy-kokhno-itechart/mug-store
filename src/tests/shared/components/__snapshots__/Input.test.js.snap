// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> check Input renders properly with custom type provided 1`] = `
<FormGroup
  error=""
  groupClasses=""
  label="Test label"
  labelClasses=""
  name="testname"
>
  <input
    className="form-control"
    id="testname"
    key="testname"
    name="testname"
    onChange={[Function]}
    type="password"
    value=""
  />
</FormGroup>
`;

exports[`<Input /> check Input renders properly with default props 1`] = `
<FormGroup
  error=""
  groupClasses=""
  label="Test label"
  labelClasses=""
  name="testname"
>
  <input
    className="form-control"
    id="testname"
    key="testname"
    name="testname"
    onChange={[Function]}
    type="text"
    value=""
  />
</FormGroup>
`;

exports[`<Input /> check Input renders properly with error 1`] = `
<FormGroup
  error="test error"
  groupClasses=""
  label="Test label"
  labelClasses=""
  name="testname"
>
  <input
    className="form-control"
    id="testname"
    key="testname"
    name="testname"
    onChange={[Function]}
    type="text"
    value=""
  />
</FormGroup>
`;

exports[`<Input /> check Input renders properly with number value provided 1`] = `
<FormGroup
  error=""
  groupClasses=""
  label="Test label"
  labelClasses=""
  name="testname"
>
  <input
    className="form-control"
    id="testname"
    key="testname"
    name="testname"
    onChange={[Function]}
    type="text"
    value={5}
  />
</FormGroup>
`;

exports[`<Input /> check Input renders properly with string value provided 1`] = `
<FormGroup
  error=""
  groupClasses=""
  label="Test label"
  labelClasses=""
  name="testname"
>
  <input
    className="form-control"
    id="testname"
    key="testname"
    name="testname"
    onChange={[Function]}
    type="text"
    value="Test value"
  />
</FormGroup>
`;
